game/theater/start_generator.py

```diff
@@ -276,314 +300,169 @@ def generate_ground_points(self) -> None:
         for position in self.control_point.preset_locations.coastal_defenses:
-            self.generate_coastal_site_at(position)
-
-    def generate_coastal_site_at(self, position: PointWithHeading) -> None:
-        group_id = self.game.next_group_id()
-
-        g = CoastalSiteGroundObject(
-            namegen.random_objective_name(),
-            group_id,
-            position,
-            self.control_point,
-            position.heading,
-        )
-        group = generate_coastal_group(self.game, g, self.faction_name)
-        g.groups = []
-        if group is not None:
-            g.groups.append(group)
-            self.control_point.connected_objectives.append(g)
-        return
-
-    def generate_strike_targets(self) -> None:
-        building_set = list(set(self.faction.building_set) - {"oil"})
-        if not building_set:
-            logging.error("Faction has no buildings defined")
-            return
-        for position in self.control_point.preset_locations.strike_locations:
-            category = random.choice(building_set)
-            self.generate_strike_target_at(category, position)
+            unit_group = self.ground_forces.random_group_for_role_and_task(
+                GroupRole.Defenses, GroupTask.Coastal
+            )
+            if not unit_group:
+                logging.error(f"{self.faction_name} has no UnitGroup for Coastal")
+                return
+            self.generate_ground_object_from_group(unit_group, position)
 
-    def generate_offshore_strike_targets(self) -> None:
-        if "oil" not in self.faction.building_set:
-            logging.error("Faction does not support offshore strike targets")
-            return
-        for position in self.control_point.preset_locations.offshore_strike_locations:
-            self.generate_strike_target_at("oil", position)
+    def generate_buildings(self) -> None:
+        for preset_location in self.control_point.preset_locations.buildings:
+            if preset_location.task is None:
+                logging.error(
+                    f"Building {preset_location.original_name} has no tasking"
+                )
+                continue
+            self.generate_building_at(preset_location.task, preset_location)
 
 
 class FobGroundObjectGenerator(AirbaseGroundObjectGenerator):
     def generate(self) -> bool:
-        self.generate_fob()
-        self.generate_armor_groups()
-        self.generate_factories()
-        self.generate_ammunition_depots()
-        self.generate_aa()
-        self.generate_ewrs()
-        self.generate_scenery_sites()
-        self.generate_strike_targets()
-        self.generate_offshore_strike_targets()
-
-        if self.faction.missiles:
-            self.generate_missile_sites()
-
-        if self.faction.coastal_defenses:
-            self.generate_coastal_sites()
-
-        return True
+        if super(FobGroundObjectGenerator, self).generate():
+            self.generate_fob()
```

Comment:
Yes but in general the preset location list for ships should always be empty for a FOB. The only difference is that FOB will generate a Building with the FOB Layout at the CP position.
I would argue here that if a Ship gets spawned at an FOB something else went really wrong. I personally say: running the generate navy for FOB will more like solve problems than create some. Because the campaign loader calculates which CP will be the one where the ships gets attached to based on the distance. So when a FOB is closer to a ship than some other CP the ship would in the previous version not be generated at all. (Also to note: this does not affect the actual position where it will be spawned! Just to which CP it will be attached).

WDYS @DanAlbert?